lib/sycamore/sycamore/transforms/partition.py

```diff
@@ -501,23 +502,20 @@ class Partition(Transform):
     def __init__(
         self, child: Node, partitioner: Partitioner, table_extractor: Optional[TableExtractor] = None, **resource_args
     ):
-        super().__init__(child, **resource_args)
-        self._partitioner = partitioner
-        self._table_extractor = table_extractor
-
-    def execute(self) -> Dataset:
-        input_dataset = self.child().execute()
-        if isinstance(self._partitioner, SycamorePartitioner):
-            if "num_gpus" in self.resource_args and self.resource_args["num_gpus"] < 0:
-                raise RuntimeError("Invalid GPU Nums!")
-
-            dataset = input_dataset.map(
-                generate_map_class_from_callable(self._partitioner.partition),
-                compute=ActorPoolStrategy(size=1),
-                **self.resource_args,
-            )
-        else:
-            dataset = input_dataset.map(generate_map_function(self._partitioner.partition))
-        if self._table_extractor:
-            dataset = dataset.map(generate_map_function(self._table_extractor.extract_tables))
-        return dataset
+        ops = []
+
+        if isinstance(partitioner, SycamorePartitioner):
```

Comment:
I hope we can turn this check into a property or method or something but out of scope of this